recipes-kernel/linux/files/0001-iot2050-add-iot2050-platform-support.patch

```diff
@@ -41,24 +45,70 @@ index b952d3e730d6..3f6fc2f8445c 100644
  dtb-$(CONFIG_ARCH_K3_J721E_SOC) += k3-j721e-common-proc-board.dtb \
  				   k3-j721e-proc-board-tps65917.dtb \
  				   k3-j721e-common-proc-board-infotainment.dtbo \
-diff --git a/arch/arm64/boot/dts/ti/k3-am65-iot2050-common.dtsi b/arch/arm64/boot/dts/ti/k3-am65-iot2050-common.dtsi
+diff --git a/arch/arm64/boot/dts/ti/k3-am65-iot2050-oldfw.dtsi b/arch/arm64/boot/dts/ti/k3-am65-iot2050-oldfw.dtsi
 new file mode 100644
-index 000000000000..c585fdfcc5cc
+index 000000000000..5b2bdc4fa311
 --- /dev/null
-+++ b/arch/arm64/boot/dts/ti/k3-am65-iot2050-common.dtsi
-@@ -0,0 +1,750 @@
++++ b/arch/arm64/boot/dts/ti/k3-am65-iot2050-oldfw.dtsi
+@@ -0,0 +1,39 @@
 +// SPDX-License-Identifier: GPL-2.0
 +/*
-+ * (C) Copyright 2019 Siemens AG
++ * (C) Copyright 2018-2020 Siemens AG
 + */
++
++/dts-v1/;
++
++/* workaround for uart issues (DMA warnings and RCU preempt) */
++
++&main_uart0 {
++	/delete-property/ dmas;
++	/delete-property/ dma-names;
++};
++
++&main_uart1 {
++	/delete-property/ dmas;
++	/delete-property/ dma-names;
++};
++
++&mcu_uart0 {
++	/delete-property/ dmas;
++	/delete-property/ dma-names;
++};
++
++/*
++ * Swap clock TISCI clock IDs between sdhci0 and sdhci1 to work
++ * around an issue in System Firmware 2019.12a (and earlier) known
++ * as SYSFW-3179.
++ */
++
++&sdhci0 {
++	clocks = <&k3_clks 48 0>, <&k3_clks 48 1>;
++	assigned-clocks = <&k3_clks 48 1>;
++	assigned-clock-rates = <142860000>;
++};
++
++&sdhci1 {
++	clocks = <&k3_clks 47 0>, <&k3_clks 47 1>;
++};
```

Comment:
the clock swapping of sdhci is only for release basic board..
not for all released board

Reply:
Checked on V01.00.02, this workaround is both for advanced and basic, see

https://github.com/siemens/meta-iot2050/blob/V01.00.02/recipes-kernel/linux/files/0019-feat-set-sdhci0-clock-frequency-to-142.86MHz.patch

```diff
diff --git a/arch/arm64/boot/dts/siemens/iot2050-advanced.dts b/arch/arm64/boot/dts/siemens/iot2050-advanced.dts
index 88b04c974294..9a83edbbeae1 100644
--- a/arch/arm64/boot/dts/siemens/iot2050-advanced.dts
+++ b/arch/arm64/boot/dts/siemens/iot2050-advanced.dts
@@ -43,9 +43,18 @@
 };
 
 &sdhci0 {
+	/*
+	 * Swap clock TISCI clock IDs between sdhci0 and sdhci1 to work
+	 * around an issue in System Firmware 2019.12a (and earlier) known
+	 * as SYSFW-3179.
+	 */
+	clocks = <&k3_clks 48 0>, <&k3_clks 48 1>;
+	assigned-clocks = <&k3_clks 48 1>;
+	assigned-clock-rates = <142860000>;
 	pinctrl-names = "default";
 	pinctrl-0 = <&main_mmc0_pins_default>;
 	bus-width = <8>;
 	non-removable;
 	ti,driver-strength-ohm = <50>;
+	disable-wp;
 };
\ No newline at end of file
diff --git a/arch/arm64/boot/dts/siemens/iot2050-common.dtsi b/arch/arm64/boot/dts/siemens/iot2050-common.dtsi
index 89f354528830..c58ae32547a2 100644
--- a/arch/arm64/boot/dts/siemens/iot2050-common.dtsi
+++ b/arch/arm64/boot/dts/siemens/iot2050-common.dtsi
@@ -585,9 +585,16 @@
 };
 
 &sdhci1 {
+	/*
+	 * Swap clock TISCI clock IDs between sdhci0 and sdhci1 to work
+	 * around an issue in System Firmware 2019.12a (and earlier) known
+	 * as SYSFW-3179.
+	 */
+	clocks = <&k3_clks 47 0>, <&k3_clks 47 1>;
 	pinctrl-names = "default";
 	pinctrl-0 = <&main_mmc1_pins_default>;
 	ti,driver-strength-ohm = <50>;
+	disable-wp;
 };
```